tests/MongoDB.Driver.Tests/AggregateFluentTests.cs

```diff
@@ -193,6 +194,302 @@ public void ChangeStream_should_add_the_expected_stage_when_options_is_null()
             result.Should().Be(1);
         }
 
+        [Fact]
+        public void Lookup_with_let_and_bsondocuments_params_should_return_the_expected_result()
+        {
+            RequireServer.Check().Supports(Feature.AggregateLet);
+
+            string databaseName = "test";
+            string ordersCollectionName = "orders";
+            string warehousesCollectionName = "warehouses";
+
+            var client = CreateClient();
+            DropCollection(client, databaseName, ordersCollectionName);
+            DropCollection(client, databaseName, warehousesCollectionName);
+
+            var ordersCollection = client.GetDatabase(databaseName).GetCollection<BsonDocument>(ordersCollectionName);
+            var warehousesCollection = client.GetDatabase(databaseName).GetCollection<BsonDocument>(warehousesCollectionName);
+
+            var orderDocuments = new[]
+            {
+                new BsonDocument { { "item", "almonds" }, { "price", 12 }, { "ordered", 2 } },
+                new BsonDocument { { "item", "pecans" }, { "price", 20 }, { "ordered", 1 } },
+                new BsonDocument { { "item", "cookies" }, { "price", 10 }, { "ordered", 60 } }
+            };
+            ordersCollection.InsertMany(orderDocuments);
+
+            var warehouseDocuments = new[]
+            {
+                new BsonDocument { { "stock_item", "almonds" }, { "instock", 120 } },
+                new BsonDocument { { "stock_item", "pecans" }, { "instock", 80 } },
+                new BsonDocument { { "stock_item", "almonds" }, { "instock", 60 } },
+                new BsonDocument { { "stock_item", "cookies" }, { "instock", 40 } },
+                new BsonDocument { { "stock_item", "cookies" }, { "instock", 80 } }
+            };
+            warehousesCollection.InsertMany(warehouseDocuments);
+
+            var pipeline = new EmptyPipelineDefinition<BsonDocument>()
```

Comment:
Rename variable lookupPipeline for consistency with API parameter name.

Do the same in other test methods.